panic when rpc error (dkron 0.10.2)

When rpc error
dkron 0.10.1
 -> not panic
dkron 0.10.2
 -> panic

- Linux / Mac (0.10.2)
```
INFO[2018-05-24T06:56:30+09:00] agent: Starting job                           job=ifconfig node=Mac
ERRO[2018-05-24T06:56:34+09:00] rpc: error dialing.                           err="dial tcp 172.21.0.11:6868: connect: network is unreachable" node=Mac server_addr="172.21.0.11:6868"
ERRO[2018-05-24T06:56:34+09:00] agent: Error on rpc.GetJob call               error="dial tcp 172.21.0.11:6868: connect: network is unreachable" node=Mac
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17d4c10]

goroutine 115 [running]:
github.com/victorcoder/dkron/dkron.(*Agent).eventLoop(0xc4202e54a0)
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:431 +0x13e0
created by github.com/victorcoder/dkron/dkron.(*Agent).Start
	/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:91 +0x180
```

- Windows (0.10.2)
```
INFO[2018-05-24T06:56:30+09:00] agent: Starting job                           job=windows node=Windows
ERRO[2018-05-24T06:56:51+09:00] rpc: error dialing.                           err="dial tcp 172.21.0.11:6868: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." node=Windows server_addr="172.21.0.11:6868"
ERRO[2018-05-24T06:56:51+09:00] agent: Error on rpc.GetJob call               error="dial tcp 172.21.0.11:6868: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." node=Windows
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xbefc87]

goroutine 89 [running]:
github.com/victorcoder/dkron/dkron.(*Agent).eventLoop(0xc0423f0540)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:431 +0x13e7
created by github.com/victorcoder/dkron/dkron.(*Agent).Start
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/agent.go:91 +0x187
```

--- a/dkron/agent.go
+++ b/dkron/agent.go
@@ -424,9 +424,25 @@ func (a *Agent) eventLoop() {
 					}).Info("agent: Starting job")
 
 					rpcc := RPCClient{ServerAddr: rqp.RPCAddr}
+
+					// There are two error types to handle here:
+					// Key not found when the job is removed from store
+					// Dial tcp error
+					// In case of deleted job or other error, we should report and break the flow.
+					// On dial error we should retry with a limit.
+					i := 0
+				RetryGetJob:
 					job, err := rpcc.GetJob(rqp.Execution.JobName)
 					if err != nil {
+						if err == ErrRPCDialing {
+							if i < 10 {
+								i++
+								goto RetryGetJob
+							}
+							log.WithError(err).Fatal("agent: A working RPC connection to a Dkron server must exists.")
+						}
 						log.WithError(err).Error("agent: Error on rpc.GetJob call")
+						continue
 					}
 					log.WithField("command", job.Command).Debug("agent: GetJob by RPC")
 

